<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="jliucx" />

<meta name="date" content="2024-07-12" />

<title>sceptre_example</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">first sceptre trial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jliucx/sceptre">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">sceptre_example</h1>
<h4 class="author">jliucx</h4>
<h4 class="date">2024-07-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Getting_started/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240712code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20240712)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240712code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240712)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_cis
</td>
<td style="text-align:left;">
output/sceptre_outputs_cis
</td>
</tr>
<tr>
<td style="text-align:left;">
/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_cis/results_run_calibration_check.rds
</td>
<td style="text-align:left;">
output/sceptre_outputs_cis/results_run_calibration_check.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/project/xuanyao/jiaming/Getting_started/output/sceptre_trans_singleton
</td>
<td style="text-align:left;">
output/sceptre_trans_singleton
</td>
</tr>
<tr>
<td style="text-align:left;">
/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_trans_singleton/results_run_calibration_check.rds
</td>
<td style="text-align:left;">
output/sceptre_outputs_trans_singleton/results_run_calibration_check.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjliucxsceptretreeef69a3762402fd2a9ab2b51e0805a5ed15ea1839targetblankef69a37a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jliucx/sceptre/tree/ef69a3762402fd2a9ab2b51e0805a5ed15ea1839" target="_blank">ef69a37</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjliucxsceptretreeef69a3762402fd2a9ab2b51e0805a5ed15ea1839targetblankef69a37a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jliucx/sceptre/tree/ef69a3762402fd2a9ab2b51e0805a5ed15ea1839" target="_blank">ef69a37</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/job-info.err
    Ignored:    analysis/job-info.out
    Ignored:    analysis/run.sh

Untracked files:
    Untracked:  analysis/sceptre_singleton.Rmd
    Untracked:  analysis/sceptre_threshold.Rmd
    Untracked:  code/create_grna_target_data_frame.Rmd
    Untracked:  code/try_Knit.Rmd
    Untracked:  data/GDO_threshold/
    Untracked:  data/STINGseq-v1_GDO/
    Untracked:  data/STINGseq-v1_HTO/
    Untracked:  data/STINGseq-v1_cDNA/
    Untracked:  output/sceptre_outputs/trans_mixture.txt
    Untracked:  output/sceptre_outputs_cis/
    Untracked:  output/sceptre_outputs_cis_singleton/
    Untracked:  output/sceptre_outputs_cis_threshold/
    Untracked:  output/sceptre_outputs_cis_threshold5/
    Untracked:  output/sceptre_outputs_trans_omit_NTC3_12/
    Untracked:  output/sceptre_outputs_trans_singleton/
    Untracked:  output/sceptre_outputs_trans_singleton_omit_NTC3_12/
    Untracked:  output/sceptre_outputs_trans_threshold/
    Untracked:  output/sceptre_outputs_trans_threshold_omit_NTC3_12/
    Untracked:  output/try_rcc/

Unstaged changes:
    Modified:   .gitignore
    Deleted:    analysis/create_grna_target_data_frame.Rmd
    Modified:   analysis/sceptre.Rmd
    Modified:   analysis/sceptre_example.Rmd
    Deleted:    analysis/seurat_example.Rmd
    Modified:   analysis/summary.Rmd
    Deleted:    analysis/try_Knit.Rmd
    Deleted:    analysis/try_rcc.Rmd
    Deleted:    output/sceptre_outputs/analysis_summary.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/qqplot.Rmd</code>) and HTML (<code>docs/qqplot.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/ef69a3762402fd2a9ab2b51e0805a5ed15ea1839/analysis/qqplot.Rmd" target="_blank">ef69a37</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-09
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/af346f4475b10c87b04ff4e17243b0bb9c8f7280/docs/qqplot.html" target="_blank">af346f4</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/3b3d02b349ab4fe16d97d678f5bd184d8cc59070/analysis/qqplot.Rmd" target="_blank">3b3d02b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-09
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/123faeb58636daaacb266bd058825c530029d821/docs/qqplot.html" target="_blank">123faeb</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/f988bf1269bfc4218ee58b5a64ced2ecfd25b5a6/analysis/qqplot.Rmd" target="_blank">f988bf1</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/c99d60f22a6738cece28c0f31bc7a51622784d9b/docs/qqplot.html" target="_blank">c99d60f</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/9377797e45162b23342ca45d3fcae2a6f72ab8a9/analysis/qqplot.Rmd" target="_blank">9377797</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/57f4a97db8ac330b865f286d403c7a5ede6a7bce/docs/qqplot.html" target="_blank">57f4a97</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/84b992457b9fe702373e101a8e504649db7d75e0/analysis/qqplot.Rmd" target="_blank">84b9924</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-07
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/f075160dba27b1aba1aabeae17ed7cdfdb2e0328/docs/qqplot.html" target="_blank">f075160</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a748c2179a076a1dc559b9f11a6d6478c98b5d5c/analysis/qqplot.Rmd" target="_blank">a748c21</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/3ec664cf94a011ce1a8c3cb3a50488364def0a2e/docs/qqplot.html" target="_blank">3ec664c</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/d1c2b02d812c13b2b8933bc50803d319b80a3988/analysis/qqplot.Rmd" target="_blank">d1c2b02</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jliucx/sceptre/ae94bb6356ef460b6d15e0b5867cc83e21249a8d/docs/qqplot.html" target="_blank">ae94bb6</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/analysis/qqplot.Rmd" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
<td>
add summary
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="manually-examine-the-negative-control-p-values" class="section level2">
<h2>Manually examine the negative control p-values</h2>
<p>We extract the results of negative control p- values of <strong>“/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_cis”</strong> (cis analysis with grna assigned by mixture model). We want to confirm that the qq plot we manually create is the same as the plot that “sceptre” package creates:</p>
<pre class="r"><code>library(qqman)</code></pre>
<pre><code></code></pre>
<pre><code>For example usage please run: vignette(&#39;qqman&#39;)</code></pre>
<pre><code></code></pre>
<pre><code>Citation appreciated but not required:</code></pre>
<pre><code>Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(ggplot2)
results_run_calibration_check &lt;- readRDS(&quot;/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_cis/results_run_calibration_check.rds&quot;)</code></pre>
<pre class="r"><code>p_val=results_run_calibration_check$p_value</code></pre>
<pre class="r"><code>p_val &lt;- sort(p_val)
n &lt;- length(p_val)
expected &lt;- ppoints(n)  

# Calculate observed p-values
observed &lt;- p_val  # Reversing the scale

# Create a data frame for plotting
data &lt;- data.frame(Expected = expected, Observed = observed)

# Create the QQ plot with reversed axes
ggplot(data, aes(x = Expected, y = Observed)) +
  geom_point(size=0.9, color=&quot;red&quot;) +  # Plot the points
  geom_abline(slope = 1, intercept = 0, color = &quot;black&quot;) +  # Add reference line
  scale_x_reverse(limits = c(1, 0)) +  # Reverse x-axis
  scale_y_reverse(limits = c(1, 0)) +  # Reverse y-axis
  coord_fixed() +
  labs(title = &quot;QQ plot (bulk)&quot;,
       x = &quot;Expected null p-value&quot;,
       y = &quot;Observed p-value&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/qqplot.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/ae94bb6356ef460b6d15e0b5867cc83e21249a8d/docs/figure/qqplot.Rmd/unnamed-chunk-3-1.png" target="_blank">ae94bb6</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gg_qqplot &lt;- function(ps, ci = 0.95) {
  n  &lt;- length(ps)
  df &lt;- data.frame(
    observed = -log10(sort(ps)),
    expected = -log10(ppoints(n)),
    clower   = -log10(qbeta(p = (1 - ci) / 2, shape1 = 1:n, shape2 = n:1)),
    cupper   = -log10(qbeta(p = (1 + ci) / 2, shape1 = 1:n, shape2 = n:1))
  )
  log10Pe &lt;- expression(paste(&quot;Expected -log&quot;[10], plain(P)))
  log10Po &lt;- expression(paste(&quot;Observed -log&quot;[10], plain(P)))
  ggplot(df) +
    geom_ribbon(mapping = aes(x = expected, ymin = clower, ymax = cupper),
                alpha = 0.1) +
    geom_point(aes(expected, observed), size = 1.5, color=&quot;red&quot;) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.5) +
    xlab(log10Pe) + ylab(log10Po)
}</code></pre>
<pre class="r"><code>gg_qqplot(p_val)</code></pre>
<p><img src="figure/qqplot.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/ae94bb6356ef460b6d15e0b5867cc83e21249a8d/docs/figure/qqplot.Rmd/unnamed-chunk-5-1.png" target="_blank">ae94bb6</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Below is the qq plot of negative control p-values, which is identical to the qq plots that we manually create.</p>
<p><img src="images/cis_plot_run_calibration_check.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cis_plot_run_calibration_check">
Past versions of cis_plot_run_calibration_check.png
</button>
</p>
<div id="fig-cis_plot_run_calibration_check" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/cis_plot_run_calibration_check.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="the-p-values-are-raw-on-the-output-file-and-subject-to-bh-correction-to-test-significance" class="section level2">
<h2>The p-values are raw on the output file, and subject to BH correction to test significance</h2>
<p>To examine whether the p-values on the summary table are subject to multiple testing correction (if so, what type of correction?), we notice that the p_value and the significant results does not match (although the significance level <span class="math inline">\(\alpha=0.1\)</span>, most p_values with magnitude e-05 or beyond are labeled “not significant”).</p>
<p><strong>The output file we used here is “/project/xuanyao/jiaming/Getting_started/output/sceptre_trans_singleton”</strong></p>
<p>According to the package description, The parameter <code>multiple_testing_method</code> controls the multiple testing method that is used to adjust the p-values. The default option is Benjamini-Hochberg (<code>"BH"</code>). The parameter <code>multiple_testing_alpha</code> (default value <code>0.1</code>) controls the level of the multiple testing procedure. If <code>multiple_testing_method</code> is set to a method that controls that false discovery rate (e.g., <code>"BH"</code> or <code>"BY"</code>), then <code>multiple_testing_alpha</code> is the nominal false discovery rate. If, on the other hand, <code>multiple_testing_method</code> is set to a method that controls the family-wise error rate (e.g., <code>"bonferroni"</code> or <code>"holm"</code>), then <code>multiple_testing_alpha</code> is the nominal family-wise error rate.</p>
<p>We create a new column “adj_p_value” which applies BH correction to the p-values. Not surprisingly, the adjusted p values matches perfectly with significant results. This indicate that the output p_values of sceptre are raw, and the significant results are based on adjusted p-values.</p>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>results_run_calibration_check &lt;- readRDS(&quot;/project/xuanyao/jiaming/Getting_started/output/sceptre_outputs_trans_singleton/results_run_calibration_check.rds&quot;)
results_run_calibration_check &lt;- results_run_calibration_check %&gt;%
  mutate(adj_p_value = p.adjust(p_value, method = &quot;BH&quot;))

results_run_calibration_check[c(1:10,70:80),c(1,2,7,9,10)]</code></pre>
<pre><code>       response_id grna_id      p_value significant  adj_p_value
1  ENSG00000111371   NTC-3 2.098900e-37        TRUE 3.265784e-32
2  ENSG00000105397   NTC-3 5.915662e-15        TRUE 4.602237e-10
3  ENSG00000134453   NTC-3 5.164750e-10        TRUE 2.678698e-05
4  ENSG00000140543   NTC-8 3.345412e-09        TRUE 1.301324e-04
5  ENSG00000148288  NTC-12 7.148804e-09        TRUE 2.224636e-04
6  ENSG00000134419   NTC-8 3.212665e-08        TRUE 8.331244e-04
7  ENSG00000239672   NTC-3 2.209987e-07        TRUE 4.912327e-03
8  ENSG00000126432   NTC-3 7.304610e-07        TRUE 1.378937e-02
9  ENSG00000148303   NTC-8 9.435981e-07        TRUE 1.378937e-02
10 ENSG00000171863  NTC-10 1.010959e-06        TRUE 1.378937e-02
70 ENSG00000103021  NTC-12 4.580678e-05       FALSE 1.007067e-01
71 ENSG00000102390   NTC-5 4.595377e-05       FALSE 1.007067e-01
72 ENSG00000160446  NTC-11 4.913498e-05       FALSE 1.061827e-01
73 ENSG00000152795   NTC-3 5.197139e-05       FALSE 1.105604e-01
74 ENSG00000142937   NTC-8 5.258184e-05       FALSE 1.105604e-01
75 ENSG00000114779   NTC-8 5.346174e-05       FALSE 1.109117e-01
76 ENSG00000160229   NTC-6 5.593555e-05       FALSE 1.134165e-01
77 ENSG00000165732   NTC-3 5.636354e-05       FALSE 1.134165e-01
78 ENSG00000149646  NTC-12 5.685587e-05       FALSE 1.134165e-01
79 ENSG00000274997   NTC-9 5.962407e-05       FALSE 1.174330e-01
80 ENSG00000165629  NTC-12 6.113559e-05       FALSE 1.187852e-01</code></pre>
</div>
<div id="individual-analysis-of-the-non-targeting-grnas" class="section level2">
<h2>**Individual analysis of the non-targeting gRNAs</h2>
<p>We noticed that the calibration check result when we perform “trans analysis”(it also contains cis analysis) is inflated. To see whether there might be problem with NT gRNA design, we find that NTC-3 and NTC-12 contribute to over 50% percent of false positive discoveries.</p>
<pre class="r"><code>filtered_data &lt;- results_run_calibration_check %&gt;% filter(significant == TRUE)
grna_count &lt;- filtered_data %&gt;%
  group_by(grna_id) %&gt;%
  summarise(count = n()) %&gt;%
  ungroup()

# Calculate the percentage
grna_count &lt;- grna_count %&gt;%
  mutate(percentage = count / sum(count) * 100)

# Create the pie chart
ggplot(grna_count, aes(x = &quot;&quot;, y = count, fill = grna_id)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;) +
  theme_void() +
  geom_text(aes(label = paste0(round(percentage, 1), &quot;%&quot;)), 
            position = position_stack(vjust = 0.5)) +
  labs(title = &quot;Distribution of significant grna_id&quot;, fill = &quot;grna_id&quot;)</code></pre>
<p><img src="figure/qqplot.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/ae94bb6356ef460b6d15e0b5867cc83e21249a8d/docs/figure/qqplot.Rmd/unnamed-chunk-8-1.png" target="_blank">ae94bb6</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Therefore, we perform the analysis again omitting the NTC-3 and NTC-12, and below are the outcomes.</p>
<div id="trans-analysis-with-grna-method-using-mixture-model-singleton" class="section level3">
<h3>Trans analysis with gRNA method using mixture model (singleton)</h3>
<p><img src="images/trans_singleton.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-trans_singleton">
Past versions of trans_singleton.png
</button>
</p>
<div id="fig-trans_singleton" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/9377797e45162b23342ca45d3fcae2a6f72ab8a9/docs/images/trans_singleton.png" target="_blank">9377797</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/trans_singleton.txt">text summary</a></p>
</div>
<div id="trans-analysis-with-grna-method-using-mixture-model-singletonomitting-ntc-312" class="section level3">
<h3>Trans analysis with gRNA method using mixture model (singleton)(omitting NTC-3&amp;12)</h3>
<p><img src="images/trans_calibration_check_mixture_omitNTC3.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-trans_calibration_check_mixture_omitNTC3">
Past versions of trans_calibration_check_mixture_omitNTC3.png
</button>
</p>
<div id="fig-trans_calibration_check_mixture_omitNTC3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/trans_calibration_check_mixture_omitNTC3.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/trans_mixture_omit.txt">text summary</a></p>
</div>
<div id="trans-analysis-with-grna-method-using-mixture-model" class="section level3">
<h3>Trans analysis with gRNA method using mixture model</h3>
<p><img src="images/plot_run_calibration_check-01.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_run_calibration_check-01">
Past versions of plot_run_calibration_check-01.png
</button>
</p>
<div id="fig-plot_run_calibration_check-01" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/f41dac8f3729765f941b1d8d12b7d75f8c2c57ad/docs/images/plot_run_calibration_check-01.png" target="_blank">f41dac8</a>
</td>
<td>
jliucx
</td>
<td>
2024-07-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/trans_mixture.txt">text summary</a></p>
</div>
<div id="trans-analysis-with-grna-method-using-mixture-model-omitting-ntc-312" class="section level3">
<h3>Trans analysis with gRNA method using mixture model (omitting NTC-3&amp;12)</h3>
<p><img src="images/trans_mixture_omit.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-trans_mixture_omit">
Past versions of trans_mixture_omit.png
</button>
</p>
<div id="fig-trans_mixture_omit" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/84b992457b9fe702373e101a8e504649db7d75e0/docs/images/trans_mixture_omit.png" target="_blank">84b9924</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/trans_mixture_omit2.txt">text summary</a></p>
</div>
<div id="trans-analysis-with-grna-method-using-threshold-method" class="section level3">
<h3>Trans analysis with gRNA method using threshold method</h3>
<p><img src="images/tr_threshold.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-tr_threshold">
Past versions of tr_threshold.png
</button>
</p>
<div id="fig-tr_threshold" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/9377797e45162b23342ca45d3fcae2a6f72ab8a9/docs/images/tr_threshold.png" target="_blank">9377797</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/tr_threshold.txt">text summary</a></p>
</div>
<div id="trans-analysis-with-grna-method-using-threshold-method-omitting-ntc-312" class="section level3">
<h3>Trans analysis with gRNA method using threshold method (omitting NTC-3&amp;12)</h3>
<p><img src="images/calibration_trans_threshold_omit.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-calibration_trans_threshold_omit">
Past versions of calibration_trans_threshold_omit.png
</button>
</p>
<div id="fig-calibration_trans_threshold_omit" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/calibration_trans_threshold_omit.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/trans_threshold_omit.txt">text summary</a></p>
<pre><code>
Attaching package: &#39;kableExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<table class="table table-striped table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Contingency Table
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Mixture(Singleton)
</th>
<th style="text-align:left;">
Mixture(Unicon)
</th>
<th style="text-align:left;">
Threshold
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
original
</td>
<td style="text-align:left;">
67/155595
</td>
<td style="text-align:left;">
292/976820
</td>
<td style="text-align:left;">
51/1027281
</td>
</tr>
<tr>
<td style="text-align:left;">
removing NTC3&amp;12
</td>
<td style="text-align:left;">
15/126552
</td>
<td style="text-align:left;">
158/652822
</td>
<td style="text-align:left;">
0/684410
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="how-p-values-are-affected-by-grna-integration-strategy-union-singleton-and-bonforonni" class="section level2">
<h2>How p-values are affected by gRNA integration strategy: union, singleton and Bonforonni</h2>
<p>In many cases, there are multiple gRNAs targeting at the same target. How to combine the information of them to perform analysis becomes important in those scenarios.</p>
<div id="union" class="section level3">
<h3>Union</h3>
<p>The default option for <code>grna_integration_strategy</code> is <code>"union"</code>. This strategy constructs a “grouped gRNA” by combining all gRNAs that target a given genomic element via a union operation; this “grouped gRNA” is then tested for association against the responses to which the element is paired. We illustrate this grouping strategy using an example. Suppose that “gRNA 1” and “gRNA 2” target the same genomic element. Suppose that “gRNA 1” is present in the cells indexed 3, 6, 9 and that “gRNA 2” is present in the cells indexed 1, 4, 6, 10. The “grouped gRNA” formed combining “gRNA 1” and “gRNA 2” via the union operation is defined to be present in the cells indexed 1, 3, 4, 6, 9, 10. (See schematic below.) This “grouped gRNA” is then tested against responses as if it were a single gRNA.</p>
<p><img src="images/4.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-4">
Past versions of 4.png
</button>
</p>
<div id="fig-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/4.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="bonforonni" class="section level3">
<h3>Bonforonni</h3>
<p>Consider a given target-response pair. Suppose that k gRNAs <span class="math inline">\(\text{gRNA}_1 \ldots\text{gRNA}_k\)</span> target this target. Moreover, suppose that, upon testing for association between each of these gRNAs and the response, we obtain gRNA-wise p-values <span class="math inline">\(p_1, \ldots, p_k\)</span>. The Bonferroni p-value is defined as follows:</p>
<p><span class="math display">\[
p = k \cdot \min \{ p_1, \ldots, p_k \}.
\]</span></p>
<p>The Bonferroni p-value is valid in the following sense: if <span class="math inline">\(p_1, \ldots, p_k\)</span> are valid (i.e., uniformly distributed) p-values, then <span class="math inline">\(p\)</span> too is a valid (i.e., uniformly distributed) p-value. (The gRNA-wise p-values <span class="math inline">\(p_1, \ldots, p_k\)</span> can be arbitrarily dependent.) Below, we state this result in a slightly more precise way and prove its correctness. "Superuniformity’’ is more general than uniformity and is sufficient for p-value validity. This proof is standard; see, e.g., the Bonferroni correction \href{<a href="https://en.wikipedia.org/wiki/Bonferroni_correction%7D%7BWikipedia" class="uri">https://en.wikipedia.org/wiki/Bonferroni_correction%7D%7BWikipedia</a> page}.</p>
</div>
<div id="singleton" class="section level3">
<h3>Singleton</h3>
<p>This method treat each gRNA as a unit instead of treating each gRNA target as a unit. Hence, suppose each target is targeted by 2 gRNAs, and if you test 100 target-gene pairs, you will get 200 p-values as a outcome, which will subject to multiple testing correction later for significance analysis.</p>
</div>
</div>
<div id="how-p-values-are-affected-by-grna-assignment-strategy-mixture-model-threshold" class="section level2">
<h2>How p-values are affected by gRNA assignment strategy: mixture model, threshold</h2>
<p>In order to find out how the assignment of gRNA to cell affect the p-value, we perform different assignment strategy and compare different calibration results.</p>
<div id="cis-analysis-with-grna-method-using-mixture-model" class="section level3">
<h3>Cis analysis with gRNA method using mixture model</h3>
<p><img src="images/cis_plot_run_calibration_check.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cis_plot_run_calibration_check">
Past versions of cis_plot_run_calibration_check.png
</button>
</p>
<div id="fig-cis_plot_run_calibration_check" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/cis_plot_run_calibration_check.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/cis_mixture.txt">text summary</a></p>
</div>
<div id="cis-analysis-with-grna-method-using-threshold-from-the-author" class="section level3">
<h3>Cis analysis with gRNA method using threshold from the author</h3>
<p><img src="images/plot_run_calibration_check_author_threshold.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_run_calibration_check_author_threshold">
Past versions of plot_run_calibration_check_author_threshold.png
</button>
</p>
<div id="fig-plot_run_calibration_check_author_threshold" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/plot_run_calibration_check_author_threshold.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/cis_threshold.txt">text summary</a></p>
</div>
<div id="cis-analysis-with-grna-method-using-default-threshold5" class="section level3">
<h3>Cis analysis with gRNA method using default threshold=5</h3>
<p><img src="images/calibration_cis_threshold5.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-calibration_cis_threshold5">
Past versions of calibration_cis_threshold5.png
</button>
</p>
<div id="fig-calibration_cis_threshold5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/calibration_cis_threshold5.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><a href="images/cis_threshold5.txt">text summary</a></p>
</div>
</div>
<div id="conditional-resampling-algorithm" class="section level2">
<h2>Conditional resampling algorithm</h2>
<p><img src="images/sceptre_illustration.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sceptre_illustration">
Past versions of sceptre_illustration.png
</button>
</p>
<div id="fig-sceptre_illustration" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/sceptre_illustration.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="images/methodology.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-methodology">
Past versions of methodology.png
</button>
</p>
<div id="fig-methodology" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jliucx/sceptre/blob/a9b748bd06164d98f038e1f7f4024f4f894f3183/docs/images/methodology.png" target="_blank">a9b748b</a>
</td>
<td>
jliucx
</td>
<td>
2024-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>My point is that the rationale for this resampling algorithm is to break the latent relationship of gRNA presence and gene expression and see whether the test statistics changes significantly. Suppose there are no direct relationship between gRNA presence and gene expression, then doing so won’t change the test statistics a lot.</p>
<p>However, I think this method resample only based on the cell covariates, neglecting the fact that the measured gRNA presence is also determined by the real presence of gRNA. Hence resample based on regression on cell covariate may also break down the correlation between measured gRNA and true gRNA presence. This kind of test works definitely when we assume that measured gRNA presence are only determined by cell covariate. I am not sure whether doing conditional randomization test, conditioning only on the cell covariate is adequate in this sense. I think including the ture gRNA presence as a latent variable when doing conditional randomization test might be better, statistically.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 8

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el8-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.4.0 knitr_1.48       dplyr_1.1.4      ggplot2_3.5.1   
[5] qqman_0.1.9      workflowr_1.7.1 

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0  xfun_0.45         bslib_0.7.0       colorspace_2.1-0 
 [5] vctrs_0.6.4       generics_0.1.3    viridisLite_0.4.2 htmltools_0.5.8.1
 [9] yaml_2.2.1        utf8_1.2.1        rlang_1.1.2       jquerylib_0.1.4  
[13] later_1.2.0       pillar_1.9.0      glue_1.6.2        withr_2.5.2      
[17] calibrate_1.7.7   lifecycle_1.0.4   stringr_1.5.1     munsell_0.5.0    
[21] gtable_0.3.0      evaluate_0.23     labeling_0.4.3    callr_3.7.6      
[25] fastmap_1.1.1     httpuv_1.6.1      ps_1.7.5          fansi_1.0.5      
[29] highr_0.11        Rcpp_1.0.12       promises_1.2.0.1  scales_1.3.0     
[33] cachem_1.0.8      jsonlite_1.8.7    systemfonts_1.1.0 farver_2.1.1     
[37] fs_1.6.3          digest_0.6.33     stringi_1.6.2     processx_3.8.4   
[41] getPass_0.2-2     rprojroot_2.0.4   grid_4.1.0        cli_3.6.1        
[45] tools_4.1.0       magrittr_2.0.3    sass_0.4.9        tibble_3.2.1     
[49] whisker_0.4.1     pkgconfig_2.0.3   MASS_7.3-60       xml2_1.3.2       
[53] svglite_2.0.0     rmarkdown_2.27    httr_1.4.2        rstudioapi_0.15.0
[57] R6_2.5.1          git2r_0.32.0      compiler_4.1.0   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
