---
title: "TATES: Efficient Multivariate Genotype-Phenotype Analysis for Genome-Wide Association Studies"
author: "jliucx"
date: "2024-07-12"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

[Link to the paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3554627/)

## Background

As GWAS analyses are generally univariate in nature, multivariate phenotypic information is usually reduced to a single composite score. This means that researchers, prior to analyses, often reduce available, originally multivariate, phenotypic information (e.g., information on multiple questions from a diagnostic interview or questionnaire, or multiple items in a test) to a single phenotypic composite score, such as a continuous sum score or binary case-control status. This practice often results in loss of statistical power to detect causal variants.

On the other hand, GV effect could be either on the latent factor, or specific to one of the phenotype. And the coreration between factor and phenotype, factor and factor, phenotype and phenotype, is unknown. Univariate conceptualisations like sum scores and case-control status result in substantial loss of power to detect underlying GVs in all the scenarios except scenario A.

```{r echo=FALSE}
knitr::include_graphics("images/tates.png", error = FALSE)
```

## Core Idea

For each component of a multivariate trait, TATES combines p-values obtained in standard univariate GWAS to acquire one trait-based p-value, while correcting for correlations between components.

For individual phenotypes characterizing a trait (e.g., items or symptoms), TATES combines the p-values obtained in standard univariate GWAS to arrive at a global trait-based p-value P~T~, while correcting for the observed correlational structure between the phenotypes. Here we show that TATES has correct false positive (type-I error) rate, and that TATES picks up both phenotype-specific genetic effects as well as genetic effects that are common to multiple correlated phenotypes. Through extensive simulations, probing a wide variety of genotype-phenotype models, we demonstrate under which circumstances TATES outperforms analyses based on sum scores and MANOVA/MultiPhen with respect to the statistical power to detect causal GVs.

```{r echo=FALSE}
knitr::include_graphics("images/tates2.png", error = FALSE)
```

```{r echo=FALSE}
knitr::include_graphics("images/tates3.png", error = FALSE)
```

**Approximation p-value correlation matrix**

The *m*×*m* correlation matrix *ρ* between the p-values is not observed in practice. We used simulation to show that matrix *ρ* can be accurately approximated through the *m*×*m* correlation matrix *r* between the phenotypes. We simulated 55 continuous standard normally distributed phenotypes whose intercorrelations ranged between −.90 and .90, and a GV (MAF = .5) that was simulated to be unrelated to the 55 phenotypes. The association between the GV and all phenotypes was tested, yielding 55 p-values, and this simulation was run 10,000 times. We then calculated, across the 10,000 simulations, the mean pair-wise correlations between the 55 phenotypes (i.e., (55\*55−55)/2 = 1485 correlations), and the mean pair-wise correlations between the p-values. Regressing the vector of correlations between the p-values on the vector of correlations between the phenotypes, we obtain the 6^th^ order polynomial *ρ* = −0.0008−0.0023*r*+0.6226*r* ^2^+0.0149*r* ^3^+0.1095*r* ^4^−0.0219*r* ^5^+0.2179*r* ^6^ (coefficient of determination R^2^ = .992), allowing accurate approximation of the correlations between the p-values from the observed correlations between the phenotypes. The thus obtained matrix *ρ* is subjected to the eigenvalue decomposition.

## Other methods

MultiPhen uses ordinal regression to regress 0/1/2-coded GVs on a collection of phenotypes of any measurement nature (i.e., continuous, dichotomous, ordinal), and applies one omnibus test to test whether all regression weights in the model are together significantly different from zero. MultiPhen has been shown to outperform MANOVA when minor allele frequency (MAF) is low and the phenotypes are case-control status or non-normally distributed continuous variables. Under most circumstances, however, MultiPhen and MANOVA yield very similar results in terms of power to detect causal GVs. A drawback of these multivariate methods is that their power depends on the specific configuration of phenotypic correlations and on the location of the GV effect (e.g., on the latent dimension, or specific to one of many correlated phenotypes). For instance, when the ideal model holds, MANOVA is decidedly less powerful than univariate analyses based on sum scores. MANOVA, however, easily outperforms the sum score approach when the GV affects only one of multiple strongly correlated variables.



